core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+  }
+
+  /**
+   * This function helps in assigning role of Playtester to guest user.
+   */
+  async assignUserToPlaytesterRole(username: string): Promise<void> {
+    await this.clickOn(editbutton);
+    await this.page.waitForSelector('.e2e-test-editor-role-names', {
+      visible: true,
+    });
+    await this.clickOn(addUserName);
+    await this.type(addUserName, username);
+    await this.clickOn(addRoleBar);
+    await this.clickOn(playtester);
+    await this.clickOn(saveRole);
+    showMessage(`${username} has been added as playtester.`);
+  }
+
+  /**
+   *Exception function to verify the setting
+   *of the exploration to Public/Private
+   */
+  async expectExplorationToBePublished(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(publishButtonDropDown);
+      const publishButton = await this.page.$(mobilePublishButton);
+      if (!publishButton) {
+        showMessage(
+          'Exploration is set to Public and is accessible to Oppia users.'
+        );
+      } else {
+        throw new Error(
+          'Exploration is set to Private and is not' +
+            ' accessible to Oppia users.'
+        );
+      }
+    } else {
+      const publishButton = await this.page.$('.e2e-test-publish-exploration');
+      if (!publishButton) {
+        showMessage(
+          'Exploration is set to Public and is accessible to Oppia users.'
+        );
+      } else {
+        throw new Error(
+          'Exploration is set to Private and is not' +
+            ' accessible to Oppia users.'
+        );
+      }
+    }
+  }
+
+  /**
+   * This function helps in adding voice artist.
+   */
+  async addVoiceArtists(voiceArtists: string[]): Promise<void> {
+    for (let i = 0; i < voiceArtists.length; i++) {
+      await this.clickOn(voiceArtistEditButton);
+      await this.clickOn(addVoiceArtistUserName);
+      await this.page.waitForSelector(
+        `${addVoiceArtistUserName}:not([disabled])`
+      );
+      await this.page.click(addVoiceArtistUserName, {clickCount: 3});
+      await this.page.keyboard.press('Backspace');
+      await this.type(addVoiceArtistUserName, voiceArtists[i]);
+      await this.clickOn(voiceArtistSaveButton);
+
+      showMessage(voiceArtists[i] + ' has been added successfully.');
+    }
+  }
+
+  /**
+   * This function helps to choose notification type.
+   */
+  async optInToEmailNotifications(): Promise<void> {
+    await this.clickOn(feedbackToggleOff);
+  }
+
+  /**
+   * Exception function to verify the choice of receiving feedback
+   * and suggestion notifications via email
+   */
+  async expectEmailNotificationsToBeActivated(): Promise<void> {
+    await this.page.waitForSelector('input[id="feedback-switch"]');
+    const input = await this.page.$('input[id="feedback-switch"]');
+
+    if (!input) {
+      throw new Error('Feedback switch input element not found.');
+    }
+    const isChecked = await input.evaluate(
```

Comment:
Usually we don't start variables with verbs. Maybe change to somethingIsChecked where "something" is a description of the UI element that is checked.